metal3-dev/local-cbo.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash -xe
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+LOGDIR=${SCRIPTDIR}/logs
+source $SCRIPTDIR/logging.sh
+
+source $SCRIPTDIR/common.sh
+source $SCRIPTDIR/utils.sh
+
+if ! which kubebuilder 2>&1 >/dev/null ; then
+    echo "Did not find kubebuilder." 1>&2
+    exit 1
+fi
+
+if ! which yq 2>&1 >/dev/null ; then
+    echo "Did not find yq" 1>&2
+    echo "Install with: pip3 install --user yq" 1>&2
+    exit 1
+fi
+
+cbo_path=$GOPATH/src/github.com/openshift/cluster-baremetal-operator
+if [ ! -d $cbo_path ]; then
+    echo "Did not find $cbo_path" 1>&2
+    exit 1
+fi
+
+# Stop the machine-api-operator so it does not try to fix the
+# deployment we are going to change.
+$SCRIPTDIR/stop-mao.sh
+
+# Scale the existing deployment down.
```

Comment:
I don't think we need to manipulate the metal3 deployment, do we? The CBO will do that for us when it comes up, right?

Reply:
I think what we discussed regarding this was that eventually we will need this script to add the annotation until the CBO code catches up to adding its own cboOwnedAnnotation. During testing CBO incrementally, we would want this script to create a dummy metal3 deployment with the cboOwnedAnnotation added to it with replicas = 0.

Reply:
Also, we do not want to remove the maoOwnedAnnotation because when both the cboOwnedAnnotation and the maoOwnedAnnotation are present, the cboOwnedAnnotation should take precedence.

Reply:
I think we're mixing 2 things together in the work this PR is doing.

For CI, we need normal dev-scripts to have a variable to control creating a completely new Deployment manifest called metal3 with the expected annotation to prevent the MAO from doing anything with metal3 at all. That will let us test bringing up a cluster with the CBO and without MAO's logic for managing metal3. We will want a separate PR for that work.

This PR is creating a separate script for developers to run locally once they have a functioning cluster, to bring up a locally built CBO for testing. That script shouldn't need to do much for now, since the CBO should immediately add the required annotation to the metal3 Deployment and the MAO should stop managing it. Eventually when the CVO is installing the CBO the script will need to stop the copy of the CBO running in the cluster to allow the local replace it.

Reply:
Ok will remove the patch command then